vowpalwabbit/csv_parser/include/vw/csv_parser/parse_example_csv.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+
+#include "vw/common/text_utils.h"
+#include "vw/core/global_data.h"
+#include "vw/core/v_array.h"
+
+#include <map>
+#include <vector>
+
+namespace VW
+{
+namespace parsers
+{
+namespace csv
+{
+int csv_to_examples(VW::workspace* all, io_buf& buf, VW::v_array<example*>& examples);
+
+class parser
+{
+public:
+  parser() = default;
+
+  int parse_csv(VW::workspace* all, io_buf& buf, VW::example* ae);
+
+private:
+  std::vector<std::string> _header;
+  uint64_t _channel_hash;
+  int label_index;
+  std::map<VW::string_view, int> multiclass_label_counter;
```

Comment:
Those string_views are all taken from the io_buf, I guess they won't got freed during the parsing, and I've also tested this feature for multiclass_label_counter, and it works fine.

https://github.com/HollowMan6/vowpal_wabbit/blob/csv/vowpalwabbit/csv_parser/src/parse_example_csv.cc#L50

However, I need your further confirmation in case that I miss something. Thanks!

Reply:
Will remove the existing multiclass_label_counter and use --named_labels functionality instead. Marking this as resolved.

Reply:
Changed VW::string_view to std::string.